From a0943bd277ed9b5ecaf1c7c1ace7590b1b6da37d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH] UX: Don't draw vertical scrollbar if it is in hidden state

If fast scrollbar exit, vertical scrollbar will be hidden,
needn't draw it

diff --git a/core/java/android/view/View.java b/core/java/android/view/View.java
index bdf4141..d9c59ca 100644
--- a/core/java/android/view/View.java
+++ b/core/java/android/view/View.java
@@ -15390,7 +15390,8 @@
             scrollCache.scrollBar.setCallback(this);
         }
 
-        if (isHorizontalScrollBarEnabled() || isVerticalScrollBarEnabled()) {
+        if (isHorizontalScrollBarEnabled() ||
+                (isVerticalScrollBarEnabled() && !isVerticalScrollBarHidden())) {
 
             if (invalidate) {
                 // Invalidate to show the scrollbars
